src/osd/PG.cc

```diff
@@ -4262,6 +4263,11 @@ void PG::scrub_process_inconsistent()
 	  }
 	}
       }
+
+      scrubber.authoritative.clear();
+      scrubber.inconsistent.clear();
+      scrubber.missing.clear();
```

Comment:
I think these should get cleared even if repair is false, since otherwise we're just wasting memory.  I think that means moving them up one level (outside the if (repair)), and dropping the if (state_test(..REPAIR)) in the caller